operator/v1/types_baremetal.go

```diff
@@ -0,0 +1,68 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+
+// Metal3ProvisioningController contains configuration used by the Provisioning service (Ironic)
+// to provision baremetal hosts.
+//
+type Metal3ProvisioningController struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec is the specification of the desired behavior of the
+	// Metal3ProvisioningController.
+	Spec Metal3ProvisioningControllerSpec `json:"spec,omitempty"`
+	// status is the most recently observed status of the
+	// Metal3ProvisioningController.
+	Status Metal3ProvisioningControllerStatus `json:"status,omitempty"`
+}
+
+// ProvisioningDHCP
+type ProvisioningDHCP struct {
+	// managementState indicates whether the DHCP service would be started within
+	// the metal3 cluster or would use an external DHCP service
+	ManagementState ManagementState `json:"managementState"`
+
+	// DHCPRange
+	DHCPRange string `json:"DHCPRange,omitempty"`
+}
+
+// IngressControllerSpec is the specification of the desired behavior of the
+// IngressController.
+type Metal3ProvisioningControllerSpec struct {
+	// provisioningInterface is the name of the network interface on a Baremetal server connected
+	// to the provisioning network.
+	ProvisioningInterface string `json:"provisioningInterface"`
+
+	// provisioningIP is the IP address assigned to the provisioningInterface for provisioning
+	// the baremetal node. This IP address should be within the provisioning subnet, and
+	// outside of the DHCP range.
+	ProvisioningIP string `json:"provisioningIP"`
+
+	// provisioningNetworkCIDR is the network on which the baremetal nodes are provisioned.
+	// The provisioningIP and the IPs in the dhcpRange all come from within this network.
+	ProvisioningNetworkCIDR string `json:"provisioningNetworkCIDR"`
+
+	ProvisioningDHCP ProvisioningDHCP `json:"provisioningDHCP"`
+}
+
+// Metal3ProvisioningControllerStatus defines the observed status of the Metal3ProvisioningController.
+type Metal3ProvisioningControllerStatus struct {
```

Comment:
you need conditions at the very least.

Reply:
Agreed. We need to put some thought into what the different ConditionStatus mean for us.